src/raftstore/store/snap.rs

```diff
@@ -857,7 +901,12 @@ mod v2 {
                     warn!("failed to load existent snapshot meta when try to build {}: {:?}",
                           s.path(),
                           e);
-                    s.delete();
+                    if !retry_delete_snapshot(deleter, key, &s) {
```

Comment:
It's possible to call `retry_delete_snaphot` only when building snapshot on region thread. It would not block the raftstore thread.